src/pybind/mgr/cephadm/module.py

```diff
@@ -804,6 +810,19 @@ def _get_pub_key(self):
     def _get_user(self):
         return 0, self.ssh_user, ''
 
+    @orchestrator._cli_read_command(
+        'cephadm set-user',
+        'name=user,type=CephString',
+        'Set user for SSHing to cluster hosts, passwordless sudo will be needed for non-root users')
+    def enable_sudo_user_mode(self, user):
+        self.set_module_option('ssh_user', user)
```

Comment:
very dangerous. Maybe we can we run `host_check(self.inventory[0])` before committing this change?

Reply:
let me know if the way i did this is what you had in mind